plasmapy/utils/exceptions.py

```diff
@@ -5,6 +5,14 @@
 Custom Errors and Warning names to improve readability
 """
 
+class PhysicsError(Exception):
+    """Error for using a deeply troubling physics value"""
+    pass
+
+class PhysicsWarning(Warning):
+    """Warning for using a mildly worrisome physics value"""
+    pass
+
```

Comment:
I like these descriptions and the idea of a program being deeply worried by your physics 😛 
I'd like to see a more precise docstring describing when PhysicsWarning is to be used. Does one use it directly or is it just a class to inherit from when generating more specific warnings?

Reply:
I'd also suggest extending the docstring.  I have never documented an exception before (or even written one!) so I looked up an [example in the numpydoc format](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html):

```Python
class ExampleError(Exception):
    """Exceptions are documented in the same way as classes.

    The __init__ method may be documented in either the class level
    docstring, or as a docstring on the __init__ method itself.

    Either form is acceptable, but the two should not be mixed. Choose one
    convention to document the __init__ method and be consistent with it.

    Note
    ----
    Do not include the `self` parameter in the ``Parameters`` section.

    Parameters
    ----------
    msg : str
        Human readable string describing the exception.
    code : :obj:`int`, optional
        Numeric error code.

    Attributes
    ----------
    msg : str
        Human readable string describing the exception.
    code : int
        Numeric error code.

    """

    def __init__(self, msg, code):
        self.msg = msg
        self.code = code
```